### PR TITLE
Gets a Message by ID

### PR DESCRIPTION
**Pull request by DanielMwendwa**

### CODE REVIEW

File: coda_v2_python_client/firebase_client_wrapper.py [hunk 1]
```diff
@@ -111,3 +112,27 @@ def get_segment_count(self, dataset_id):
...
+        """
+        return self._client.document(f"datasets/{segment_id}/messages/{message_id}")
+
+    def get_message(self, dataset_id, message_id):
+        """Gets a message from a dataset by id.
+
+        :return: A message from a dataset.
+        :rtype: core_data_modules.data_models.message.Message | None
+        """
+        raw_message = self.get_message_ref(dataset_id, message_id).get().to_dict()
```
Comment (as2388) [inline coda_v2_python_client/firebase_client_wrapper.py:134]: Can you confirm you tested this on a multi-segment dataset? 

This doesn't look right as-is, as you are only checking the first segment. You actually need to check all of them until you find a match

Comment (DanielMwendwa) [reply coda_v2_python_client/firebase_client_wrapper.py]: Sorry I didn't spot that. Thanks I've updated.

File: coda_v2_python_client/firebase_client_wrapper.py [hunk 2]
```diff
@@ -111,3 +112,60 @@ def get_segment_count(self, dataset_id):
...
+                segment_id = self.id_for_segment(dataset_id, segment_index)
+                message = self.get_segment_message(segment_id, message_id)
+                if message is not None:
+                    log.debug(f"Message found in segment {segment_id}")
+                    break
+                log.debug(f"Message not found in segment {segment_id}")
+            else:
+                log.debug(f"Message not found in Dataset {dataset_id}")
+
+        return message
```
Comment (as2388) [inline coda_v2_python_client/firebase_client_wrapper.py:171]: Consider returning immediately if you find a match. Then you (i) don't need the relatively obscure for-else feature, and (ii) don't have to keep track of all the things that message might be when it's returned at the end. The final statement would simply be return None.

```suggestion
        if segment_count is None or segment_count == 1:
            return self.get_segment_message(dataset_id, message_id)
       
        log.info(f"Checking for message with ID {message_id} in {segment_count} segments of {dataset_id}...")
        for segment_index in range(1, segment_count + 1):
            segment_id = self.id_for_segment(dataset_id, segment_index)
            message = self.get_segment_message(segment_id, message_id)
            if message is not None:
                log.debug(f"Message found in segment {segment_id}")
                return message

        return None
```

File: coda_v2_python_client/firebase_client_wrapper.py [hunk 3]
```diff
@@ -111,3 +112,57 @@ def get_segment_count(self, dataset_id):
...
+        :param message_id: Id of a message.
+        :type message_id: str
+        :return: A message from a dataset.
+        :rtype: core_data_modules.data_models.message.Message | None
+        """
+        segment_count = self.get_segment_count(dataset_id)
+        if segment_count is None or segment_count == 1:
+            return self.get_segment_message(dataset_id, message_id)
+       
+        log.info(f"Checking for message with ID {message_id} in {segment_count} segments of {dataset_id}...")
```
Comment (as2388) [inline coda_v2_python_client/firebase_client_wrapper.py:160]: I think this ought to be debug, if included at all. If you keep this, can you add a log for the non-segmented case too for consistency in the logs?

**Review by as2388 — APPROVED** [review]

File: coda_v2_python_client/firebase_client_wrapper.py (hunk 3 above)
Comment (DanielMwendwa) [reply coda_v2_python_client/firebase_client_wrapper.py:160]: Thought about it and for some reason, I will not include it.

**Pull request merged**